src/server/ApiRoutes/Hold.js

```diff
@@ -54,24 +68,85 @@ function postHoldAPI(req, pickedUpItemId, cb, errorCb) {
     .catch(errorCb);
 }
 
+/**
+ * getDeliveryLocations(bibData, barcode, patronId, accessToken, cb)
+ * The function to make a request to get delivery locations of an item.
+ *
+ * @param {object} bibData
+ * @param {string} barcode
+ * @param {string} patronId
+ * @param {string} accessToken
+ * @param {function} callback
+ * @return {function}
+ */
+function getDeliveryLocations(bibData, barcode, patronId, accessToken, cb) {
+  return axios.get(
+    `${apiBase}/request/deliverylocationsbybarcode?barcodes[]=${barcode}&patronId=${patronId}`,
+    { headers:
+      {
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${accessToken}`,
+      },
+    }
+  )
+  .then(barcodeAPIresponse => {
+    cb(
+      bibData,
+      barcodeAPIresponse.data.itemListElement[0].deliveryLocation,
+      barcodeAPIresponse.data.itemListElement[0].eddRequestable
+    );
+  })
+  .catch(barcodeAPIError => {
+    console.error(
+      `deliverylocationsbybarcode API error: ${JSON.stringify(barcodeAPIError, null, 2)}`
+    );
+  });
+}
+
+/**
+ * confirmRequestServer(req, res, next)
+ * The function to return the bib and item data with its delivery locations to confirmation page.
+ *
+ * @param {req}
+ * @param {res}
+ * @param {next}
+ * @return {function}
+ */
 function confirmRequestServer(req, res, next) {
   const bibId = req.params.bibId || '';
   const loggedIn = User.requireUser(req, res);
+  const error = req.query.error ? JSON.parse(req.query.error) : {};
 
   if (!loggedIn) return false;
 
+  const accessToken = req.tokenResponse.accessToken || '';
+  const patronId = req.tokenResponse.decodedPatron.sub || '';
+  let barcode;
+
+  // Retrieve item
   return Bib.fetchBib(
     bibId,
-    (data) => {
-      res.locals.data.Store = {
-        bib: data,
-        searchKeywords: '',
-        error: {},
-      };
-      next();
+    (bibResponseData) => {
+      barcode = LibraryItem.getItem(bibResponseData, req.params.itemId).barcode;
+
+      getDeliveryLocations(
+        bibResponseData,
```

Comment:
Remove `bibResponseData` as a parameter to this function. You are passing the data from `fetchBib` to the `getDeliveryLocations` function and then in the `getDeliveryLocations` function you are passing it back to the callback function as `bibData`. From here it is being added to the `Store` as `bib: bibData` (line 139). So you can remove that unnecessary back and forth of data and just do `bib: bibResponseData` (line 139).